graphix/pattern.py

```diff
@@ -674,48 +674,132 @@ def _measurement_order_space(self):
+        G.add_edges_from(edges)
+        isolated = list(nx.isolates(G))
+        if isolated:
+            raise ValueError("The input graph must be connected")
+        meas_plane = self.get_meas_plane()
+        g, l_k = gflow(G, set(), set(self.output_nodes), meas_plane=meas_plane)
+        if not g:
+            raise ValueError("No gflow found")
+        k, layers = get_layers(l_k)
+        meas_order = []
+        while k > 0:
+            for node in layers[k]:
+                meas_order.append(node)
+            k -= 1
+        return meas_order
+
     def sort_measurement_commands(self, meas_order):
         """Convert measurement order to sequence of measurement commands
 
         Parameters
-        --------
-        meas_order: list of ints
+        ----------
+        meas_order: list of int
             optimal measurement order.
 
         Returns
-        --------
-        meas_flow: list of commands
+        -------
+        meas_cmds: list of command
             sorted measurement commands
         """
-        meas_flow = []
+        meas_cmds = []
         for i in meas_order:
             target = 0
             while True:
                 if (self.seq[target][0] == "M") & (self.seq[target][1] == i):
-                    meas_flow.append(self.seq[target])
+                    meas_cmds.append(self.seq[target])
                     break
                 target += 1
-        return meas_flow
+        return meas_cmds
 
-    def get_measurement_order(self):
-        """Returns the list containing the node indices,
+    def get_measurement_commands(self):
+        """Returns the list containing the measurement commands,
         in the order of measurements
 
         Returns
         -------
-        meas_flow : list
-            list of node indices in the order of meaurements
+        meas_cmds : list
+            list of measurement commands in the order of meaurements
         """
         if not self.is_standard():
             self.standardize()
-        meas_flow = []
+        meas_cmds = []
         ind = self._find_op_to_be_moved("M")
         if ind == "end":
             return []
         while self.seq[ind][0] == "M":
-            meas_flow.append(self.seq[ind])
+            meas_cmds.append(self.seq[ind])
             ind += 1
-        return meas_flow
+        return meas_cmds
+
+    def get_meas_plane(self):
+        """get measurement plane from the pattern.
+
+        Returns
+        -------
+        meas_plane: dict of str
+            list of str representing measurement plane for each node.
+        """
+        meas_plane = dict()
+        XYtoXZ_list = [
+            6,
+            8,
+            11,
+            12,
+            20,
+            21,
+            22,
+            23,
+        ]  # clifford indices that change meas plane from xy to xz
+        for cmd in self.seq:
+            if cmd[0] == "M":
+                if len(cmd) == 7:
+                    if cmd[6] in XYtoXZ_list:
```

Comment:
don't we need checks of measurement plane in the measurement command?

Reply:
I updated this method for general measurement planes and also added test for it.